mabwiser/utils.py

```diff
@@ -60,3 +62,148 @@ def reset(dictionary: Dict, value) -> NoReturn:
+        """
+        pass
+
+    @abc.abstractmethod
+    def choice(self, a: Union[int, Iterable[int]], size: Union[int, Tuple[int]] = None, p: Iterable[float] = None):
+        """ Return a random sample from a given 1-D array
+            Based on the probabilities associated with each entry in the array
+
+            Parameters
+            ----------
+            a : 1-D array-like or int
+                If an ndarray, a random sample is generated from its elements.
+                If an int, the random sample is generated as if a were np.arange(a)
+            size : int or tuple of ints, optional
+                    Output shape.  If the given shape is, e.g., ``(m, n, k)``, then
+                    ``m * n * k`` samples are drawn.  Default is None, in which case a
+                    single value is returned.
+            p : 1-D array-like, optional
+                The probabilities associated with each entry in options.
+
+            Returns
+            -------
+            samples : single item or ndarray
+                The generated random samples
+        """
+
+        pass
+
+    @abc.abstractmethod
+    def beta(self, alpha: int, beta: int):
+        """ Return a sample from a Beta distribution.
+
+            Parameters
+            ----------
+            alpha : float or array_like of floats
+                    Alpha, positive (>0).
+            beta : float or array_like of floats
+                    Beta, positive (>0).
+
+            Returns
+            -------
+            out : scalar
+                Drawn sample from the parameterized beta distribution.
+        """
+        pass
+
+    @abc.abstractmethod
+    def standard_normal(self, size):
+        """ Draw samples from a standard Normal distribution (mean=0, stdev=1).
+
+            Parameters
+            ----------
+            size : int or tuple of ints
+                Output shape.  If the given shape is, e.g., ``(m, n, k)``, then
+                ``m * n * k`` samples are drawn.
+
+            Returns
+            -------
+            out : ndarray
+                A floating-point array of shape ``size`` of drawn samples
+        """
+        pass
+
+
+class _NumpyRNG(_BaseRNG):
+
+    def __init__(self, seed):
+        super().__init__(seed)
+        self.rng = np.random.RandomState(self.seed)
+
+    def rand(self):
+        return self.rng.rand()
+
+    def randint(self, low: int, high: int = None, size: int = None):
+        return self.rng.randint(low=low, high=high, size=size)
+
+    def choice(self, a: Union[int, Iterable[int]], size: Union[int, Tuple[int]] = None, p: Iterable[float] = None):
+        return self.rng.choice(a=a, size=size, p=p)
+
+    def beta(self, num_success: int, num_failure: int):
+        return self.rng.beta(num_success, num_failure)
+
+    def standard_normal(self, size):
+        return self.rng.standard_normal(size)
+
+
+def _create_rng(seed: int) -> _BaseRNG:
```

Comment:
If the requirement is that a _BaseRNG object be passed to BaseMAB constructors, this should be public

Reply:
Agreed! I made it public first, but could not find any usage so downgraded to private. 

You mentioned customized bandits, which would need this functionality! So I will make it public and update the examples for customized bandits.

Reply:
See my comment about this below. On a second look, I don't see why it should be public now. The examples definitely need to be updated to point out how to use _BaseRNG but just like how it uses _Greedy, _Linear it can use private functionality. I doubt we want this in the public API, since it is not used. 

Thoughts?